plugins/fileservices/fileservices.cpp

```diff
@@ -2848,19 +2841,11 @@ FILESERVICES_API void  FILESERVICES_CALL fsDeleteExternalFile(ICodeContext * ctx
             throw makeStringExceptionV(-1,"Cannot resolve location %s",location);
         CDfsLogicalFileName lfn;
         lfn.setExternal(location,path);
+        checkExternalFileRights(ctx,lfn.get(),false,true);
```

Comment:
re. this previous comment : https://github.com/hpcc-systems/HPCC-Platform/pull/18275#discussion_r1485986966

It looks like it now, passes the full filename here (e.g. scope1::scope2::filename), then scope checks it, including "filename" tail.

The previous comment was suggesting, that since you do not know whether the tail is a directory or a filename, you should assume it's a filename and scope check the parent ("scope1::scope2"), only then once you have proved the user has access to that scope, should you check if the tail ("filename" in this case) is a directory, if it is, you need to scope check that too, if not, you are done.